components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java

```diff
@@ -4285,6 +4285,38 @@ public static String getAuthServerURL() throws APIManagementException {
     }
 
     /**
+     * This method will return gateway endpoints in a comma separated string.
+     *
+     * @param api API will used to get published environments
+     * @return endpoints comma separated string
+     * @throws APIManagementException If API key manager URL unspecified
+     */
+    public static String getGatewayEndpoints(API api) throws APIManagementException {
+        String endpoints = "none";
+        APIManagerConfiguration config = ServiceReferenceHolder.getInstance().
+                getAPIManagerConfigurationService().getAPIManagerConfiguration();
+        String publishedEnvironments = writeEnvironmentsToArtifact(api);
+        String[] publishedEnvironmentSet = publishedEnvironments.split(",");
+        for (int i = 0; i < config.getApiGatewayEnvironments().keySet().toArray().length; i++) {
+            for (int j = 0; j < publishedEnvironmentSet.length; j++) {
+                if (config.getApiGatewayEnvironments().keySet().toArray()[i].toString()
+                        .equals(publishedEnvironmentSet[j])) {
+                    Environment environment = (Environment) config.getApiGatewayEnvironments().values().toArray()[i];
+                    endpoints = environment.getApiGatewayEndpoint();
+                    break;
+                }
+            }
+        }
+        if (StringUtils.isBlank(publishedEnvironments)) {
```

Comment:
shouldn't these exception throwing come before the for loop. so why should be return exception, can we simply send there is no gateway endpoints or similar thing without throwing exception :sunglasses: